eldritch/test_encounters.py

```diff
@@ -520,6 +624,343 @@ def testSociety4Fail(self):
+      choice = self.resolve_to_choice(MultipleChoice)
+    choice.resolve(self.state, "Stamina")
+    self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 4)
+    self.assertEqual(self.char.sanity, 3)
+
+  def testHouse7Poor(self):
+    self.state.event_stack.append(encounters.House7(self.char))
+    self.char.dollars = 2
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.stamina, 3)
+    self.assertEqual(self.char.sanity, 3)
+
+  def testHouse7No(self):
+    self.state.event_stack.append(encounters.House7(self.char))
+    self.char.dollars = 5
+    choice = self.resolve_to_choice(MultipleChoice)
+    choice.resolve(self.state,"No")
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 5)
+    self.assertEqual(self.char.stamina, 3)
+    self.assertEqual(self.char.sanity, 3)
+
+  def testHouse7Yes0(self):
+    self.state.event_stack.append(encounters.House7(self.char))
+    self.char.dollars = 5
+    self.char.sanity = 1
+    self.char.stamina = 1
+    choice = self.resolve_to_choice(MultipleChoice)
+    choice.resolve(self.state,"Yes")
+    choice2 = self.resolve_to_choice(MultipleChoice)
+    choice2.resolve(self.state, 0)
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.stamina, 5)
+    self.assertEqual(self.char.sanity, 1)
+
+  def testHouse7Yes1(self):
+    self.state.event_stack.append(encounters.House7(self.char))
+    self.char.dollars = 5
+    self.char.sanity = 1
+    self.char.stamina = 1
+    choice = self.resolve_to_choice(MultipleChoice)
+    choice.resolve(self.state,"Yes")
+    choice2 = self.resolve_to_choice(MultipleChoice)
+    choice2.resolve(self.state, 1)
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.stamina, 4)
+    self.assertEqual(self.char.sanity, 2)
+
+  def testHouse7Yes2(self):
+    self.state.event_stack.append(encounters.House7(self.char))
+    self.char.dollars = 5
+    self.char.sanity = 1
+    self.char.stamina = 1
+    choice = self.resolve_to_choice(MultipleChoice)
+    choice.resolve(self.state,"Yes")
+    choice2 = self.resolve_to_choice(MultipleChoice)
+    choice2.resolve(self.state, 2)
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.stamina, 3)
+    self.assertEqual(self.char.sanity, 3)
+
+  def testHouse7Yes3(self):
+    self.state.event_stack.append(encounters.House7(self.char))
+    self.char.dollars = 5
+    self.char.sanity = 1
+    self.char.stamina = 1
+    choice = self.resolve_to_choice(MultipleChoice)
+    choice.resolve(self.state,"Yes")
+    choice2 = self.resolve_to_choice(MultipleChoice)
+    choice2.resolve(self.state, 3)
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.stamina, 2)
+    self.assertEqual(self.char.sanity, 4)
+
+  def testHouse7Yes4(self):
```

Comment:
You don't actually have to test each choice here - the unit tests for SplitGain should be enough.